src/software/ai/fsm_diagram_generator.py

```diff
@@ -59,7 +59,7 @@ def remove_suffix(str, suffix):
         # Remove suffixes from states, guard, action
         src_state = remove_suffix(src_state, "_S")
         dest_state = remove_suffix(dest_state, "_S")
-        guard = remove_suffix(guard, "_G")
+        guard = "&&".join([remove_suffix(g, "_G") for g in guard.split("&&")])
```

Comment:
Can we try doing `" && ".join`

Reply:
Good idea, that'll look nicer visually. Done.